module/enrichers.mjs

```diff
@@ -823,6 +903,36 @@ async function rollAction(event) {
         case "tool":
           options.ability = ability;
           return await actor.rollToolCheck(tool, options);
+        case "item":
+          if ( target.dataset.rollItemUuid ) {
+            const gameActor = await fromUuid(target.dataset.rollItemActor);
+            if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+              return (await fromUuid(target.dataset.rollItemUuid)).use();
+            } else {
+              return ui.notifications.warn(`You do not have ownership of ${gameActor?.name}, and cannot use this item.`);
+            }
+          } else if ( target.dataset.rollItemName ) {
+            if ( target.dataset.rollItemActor ) {
+              const gameActor = await fromUuid(target.dataset.rollItemActor);
+              if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+                if ( canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName) ) {
```

Comment:
If you defined `let item;` up top, you could simplify these, and just call `item.use()` at the very end (assuming `item` by that point has been defined).